cli/EAP7-1100-Add_the_ability_for_the_CLI_SSL_security_commands_to_be_able_to_obtain_a_server_certificate_from_Lets_Encrypt.adoc

```diff
@@ -0,0 +1,154 @@
+for enabling one-way and two-way SSL easily.
+To generate the server certificate,
+these high-level security commands made use of key-store management operations to generate a self-signed certificate.
+Since it is now possible to obtain certificates from the Let's Encrypt certificate authority using the CLI,
+the high-level security commands should be updated so that in addition to being able to generate a self-signed server certificate,
+they can also obtain a server certificate from Let's Encrypt.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-1100[EAP7-1100]
+* https://issues.jboss.org/browse/WFCORE-4227[WFCORE-4227]
+
+=== Related Issues
+
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+* mailto:jdenise@redhat.com[Jean-François Denise]
+
+=== QE Contacts
+
+* mailto:mchoma@redhat.com[Martin Choma]
+
+=== Affected Projects or Components
+
+* CLI
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+* Add parameter '--lets-encrypt' to the 'enable-ssl-http-server' and 'enable-ssl-management' security commands
+* The parameter '--lets-encrypt' should be used in combination with the '--interactive' parameter
+* After '--lets-encrypt parameter' is used user is asked for several information needed to obtain the certificate:
+** Account Key-store file name
+** Account Key-store password
+** Key-store file name
+** Key-store password
+** Certificate account name
+** Contact url(s)
+** Domain name(s)
+** Certificate alias
+** Do you agree to Let's Encrypt terms of service?
+** Enable SSL Mutual Authentication
+* Introduce '--ca-account=<name>' parameter to reuse existing --ca-account
+** This should be used mainly in combination with --interactive and --lets-encrypt parameters
+** User won't be asked to provide following items since they will be reused:
+*** Account Key-store file name
+*** Account Key-store password
+*** Certificate account name
+*** Contact url(s)
+*** Do you agree to Let's Encrypt terms of service?
+
+
+==== Example of the certificate interactive setup:
+[source,bash]
+----
+[standalone@localhost:9990 /] security enable-ssl-management --interactive --lets-encrypt
+Please provide required pieces of information to enable SSL:
+
+Let's Encrypt account key-store:
+File name (default accounts.keystore.jks):
+Password (blank generated):
+
+Let's Encrypt certificate authority account:
+Account name [admin]:
+Contact email [admin@admin.com]:
+Password (blank generated):
+
+Let's Encrypt TOS (https://community.letsencrypt.org/tos)
+Do you agree to Let's Encrypt terms of service? y/n (blank n): y
+
+Certificate info:
+Key-store file name (default management.keystore):
+Password (blank generated):
+Domain name (must be accessible by the Let's Encrypt server at 80 & 443 ports) [example.com]:
+Alias (blank generated):
+
+Enable SSL Mutual Authentication y/n (blank n): n
+
+Let's Encrypt options:
```

Comment:
I would make the summary of Let's encrypt more compact:
Account key store: accounts.keystore.jks
Password:xxx
Account keystore file X will be generated in server configuration directory.
Let's Encrypt Certificate authority account name: admin
Contact email: admin@admin.com
Password:xxxx
You provided agreement to Let's Encrypt terms of service.

Reply:
I think that we should exit the workflow. Seems strange to force the user to answer y.

Reply:
Can by the case when a user doesn't agree with the TOS written into the document, please?
And what will happen when a user doesn't agree with the TOS too?